src/Compilers/Test/Resources/Core/NetFX/Minimal/mincorlib.cs

```diff
@@ -2,8 +2,8 @@
 
 namespace System
 {
-	public class Object {}
-    public struct Byte {}
+    public class Object { }
```

Comment:
No the types and members are **not required** to exist. But if special types/members exist they are expected to be defined in a corlib. Any library without references, which defines System.Object is a valid corlib.